src/main/java/mate/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package mate.service;
+
+import java.util.Optional;
+import mate.dao.DriverDao;
+import mate.exception.AuthenticationException;
+import mate.lib.Inject;
+import mate.lib.Service;
+import mate.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private DriverDao driverDao;
+
+    public Driver login(String login, String password) throws AuthenticationException {
+        Optional<Driver> driver = driverDao.findByLogin(login);
+        if (driver.isEmpty()) {
+            throw new AuthenticationException("Login or password is invalid");
```

Comment:
Nope, we don't want to specify which particular parameter is invalid.
I use 2 consequent if blocks here instead of a single if with &&.
it doesn't change algorithm complexity but adds a few unnecessary lines.
Gonna change it)